client/server/bundler/node-moduler.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * External Dependencies
+ */
+const fs = require( 'fs' ); // eslint-disable-line  import/no-nodejs-modules
+const { builtinModules } = require( 'module' );
+const rcopy = require( 'recursive-copy' );
+const mkdirp = require( 'mkdirp' );
+
+function getModule( request ) {
+	const parts = request.split( '/' );
+	if ( parts[ 0 ].startsWith( '@' ) ) {
+		return parts[ 0 ] + '/' + parts[ 1 ];
+	}
+	return parts[ 0 ];
+}
+
+module.exports = class NodeModuler {
+	apply( compiler ) {
+		compiler.hooks.afterEmit.tapAsync( 'NodeModuler', ( compilation ) => {
```

Comment:
Does this run after applying webpack aliases and module replacements?

Reply:
Yes, module aliases and replacements are applied very early, when building the module dependency graph (starting with the entrypoint and descending recursively into dependencies).

On the other hand, `afterEmit` happens very late, after all the JS assets were compiled, optimized and written to the disk.